connector/spanmetricsconnector/config.go

```diff
@@ -70,6 +74,23 @@ type Config struct {
 	Namespace string `mapstructure:"namespace"`
 }
 
+// Validate checks if the processor configuration is valid
+func (c Config) Validate() error {
+	err := validateDimensions(c.Dimensions, dropSanitizationGate.IsEnabled())
```

Comment:
There will be some conflicts with this PR here, we need to watch the order. https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/18754

Reply:
Yes, any of those can be merged and another one adjusted afterward.